File: Math/codeforces_2004A.cpp

// Problem link ===>> https://codeforces.com/contest/2004/problem/A
// submission link ===>> https://codeforces.com/contest/2004/submission/277322375

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;
        vector<ll> a(n);
        for (ll i = 0; i < n; i++)
        {
            cin >> a[i];
        }

        if (n > 2)
        {
            cout << "NO" << endl;
        }
        else
        {
            if (abs(a[0] - a[1]) == 1)
            {
                cout << "NO" << endl;
            }
            else
            {
                cout << "YES" << endl;
            }
        }
    }
    return 0;
}

File: Math/codeforces_2004A.py

# Problem link ===>> https://codeforces.com/contest/2004/problem/A


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        x = list(map(int, input().split()))
        if n > 2 or x[0] + 1 == x[-1]:
            print('NO')
        else:
            print('YES')
        


if __name__ == "__main__":
    main()
